barista/src/main/java/com/markelliot/barista/Server.java

```diff
@@ -52,15 +52,20 @@ public final class Server {
     private final GracefulShutdownHandler shutdownHandler;
     private final Undertow undertow;
 
+    private volatile boolean started = false;
```

Comment:
Volatile is harmless but unnecessary because you're already synchronizing on the Server object when you access it.